recipes/tgbot/all/conanfile.py

```diff
@@ -24,69 +29,71 @@ class TgbotConan(ConanFile):
+        self.requires("libcurl/[>=7.78 <9]", transitive_headers=True, transitive_libs=True)
+        self.requires("openssl/[>=1.1 <4]")
 
     @property
     def _required_boost_components(self):
         return ["system"]
 
     def validate(self):
         if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, 11)
-        miss_boost_required_comp = any(getattr(self.options["boost"], "without_{}".format(boost_comp), True) for boost_comp in self._required_boost_components)
-        if self.options["boost"].header_only or miss_boost_required_comp:
-            raise ConanInvalidConfiguration("{0} requires non header-only boost with these components: {1}".format(self.name, ", ".join(self._required_boost_components)))
+            check_min_cppstd(self, 11)
+        miss_boost_required_comp = any(
+            self.dependencies["boost"].options.get_safe(f"without_{boost_comp}", True)
+            for boost_comp in self._required_boost_components
+        )
+        if self.dependencies["boost"].options.header_only or miss_boost_required_comp:
+            raise ConanInvalidConfiguration(
+                f"{self.name} requires non header-only boost with these components: "
+                + ", ".join(self._required_boost_components)
+            )
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["ENABLE_TESTS"] = False
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
 
     def _patch_sources(self):
         # Don't force PIC
-        tools.replace_in_file(
-            os.path.join(self._source_subfolder, "CMakeLists.txt"),
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "CMakeLists.txt"),
             "set_property(TARGET ${PROJECT_NAME} PROPERTY POSITION_INDEPENDENT_CODE ON)",
-            ""
+            "",
         )
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["ENABLE_TESTS"] = False
-        self._cmake.configure()
-        return self._cmake
-
     def build(self):
         self._patch_sources()
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
         cmake.install()
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        copy(self, "LICENSE",
+             dst=os.path.join(self.package_folder, "licenses"),
+             src=self.source_folder)
+        fix_apple_shared_install_name(self)
 
     def package_info(self):
         self.cpp_info.libs = ["TgBot"]
```

Comment:
```suggestion
        self.cpp_info.libs = ["TgBot"]
        self.cpp_info.defines = ["HAVE_CURL=1"]
```

Newer and older versions need to define curl support. 

https://github.com/reo7sp/tgbot-cpp/blob/v1.7.2/include/tgbot/net/CurlHttpClient.h#L4

https://github.com/reo7sp/tgbot-cpp/blob/v1.6/include/tgbot/net/CurlHttpClient.h#L4